garden_ai/garden.py

```diff
@@ -79,3 +81,65 @@ def _authenticate(self):
             on_refresh=self.auth_key_store.on_refresh,
         )
         return authorizer
+
+    def create_garden(self, **kwargs):
+        """Initialize and return a new Garden object
+
+        If invoked with keyword arguments corresponding to field names (e.g.
+        `title` or `creator`) are passed directly to the `Garden` model, making
+        it possible for the user to provide necessary info upfront to be parsed
+        and validated at initialization, or to incrementally provide field
+        values via simple assignment to attributes of the returned object
+        directly.
+
+        Parameters
+        ----------
+        **kwargs :
+            Metadata for the new Garden object. Keyword arguments matching
+            required or recommended fields will be coerced to the appropriate
+            type, if necessary.
+
+        Examples
+        --------
+
+            gc = GardenClient()
+            pea_garden = gc.create_garden(
+                creators=["Mendel, Gregor"], title="Experiments on Plant Hybridization"
+            )
+            pea_garden.year = 1863
+        """
+        return Garden(**kwargs)
+
+    def register_garden(self, G: Garden):
+        """Finalize the setup of a Garden object with user-populated fields
+
+        Parameters
+        ----------
+        G : Garden
+
+        Raises
+        ------
+        TypeError
+
+        Examples
+        --------
+        TODO
+        """
+        def request_doi(garden):
+            # TODO
+            # to be replaced with code to build the actual doi
+            return "10.55555/fake-doi"
+        try:
+            # attempts to populate a dummy garden with the data from the real
+            # one, so that it invokes our custom validators on every field (in
+            # particular, those which still have TempField defaults)
+            G.doi = request_doi(G)
+            _ = Garden(**G.dict())
```

Comment:
please add an inline comment about what is being accomplished here. the existing `Garden` metadata is being unpacked to create a new `Garden` object? feels funny

Reply:
yeah I agree it does feel funny. that's what the few lines of comments above are failing  to totally explain -- the way the pydantic model is currently set up, we're allowed to build a blank object that has invalid default values (taken from the `TempField` enum in model.py) but only checks for validity when something tries to explicitly set them. 

So if someone has a Garden they _think_ they've finished but they never gave a title, this would look to pydantic like someone tried to construct a dummy Garden that explicitly sets `title=TempField.REQUIRED`, which would implicitly trigger whatever helpful pydantic errors/exceptions we want it to

tldr: one-liner that forces a call to every validator, in case there are still fields that haven't been validated yet